metrics/tripperware.go

```diff
@@ -21,9 +22,25 @@ func Tripperware(reporter Reporter) httpwares.Tripperware {
 			tracker := reporter.Track(req)
 			start := time.Now()
 			tracker.RequestStarted()
-			req.Body = wrapBody(req.Body, func(size int) {
-				tracker.RequestRead(time.Since(start), size)
-			})
+
+			reqSize := 0
+			if req.Body != nil {
+				req.Body = wrapBody(req.Body, func(size int) {
+					reqSize = size
+				})
+			}
+			trace := httptrace.ContextClientTrace(req.Context())
```

Comment:
Can we add simple comment here that we are kind of `Listening for wroteRequest event thanks of httptrace to get exact moment of request end, because body can be nil`?